aw_watcher_window/main.py

```diff
@@ -128,6 +130,11 @@ def heartbeat_loop(client, bucket_id, poll_time, strategy, exclude_title=False):
         if current_window is None:
             logger.debug("Unable to fetch window, trying again on next poll")
         else:
+            patterns = [re.compile(re.escape(title), re.IGNORECASE) for title in exclude_titles]
```

Comment:
You are still compiling on every (outer) loop. You need to pass the compiled regexes to the heartbeat_loop function.